flox/core.py

```diff
@@ -743,7 +743,11 @@ def chunk_reduce(
 
     assert by.ndim <= array.ndim
 
-    final_array_shape = array.shape[:-nax] + (1,) * (nax - 1)
```

Comment:
I'd prefer a `chunk_accumulate` instead. Hopefully there won't be too much duplication.

Reply:
chunk reduce is used every! 
I'll try out something like:
```
def chunk_aggregate():
  if is_accumulate:
    chunk_accumulate
    ....
```

Reply:
I think I got what you meant now. Now with lots of copy/pasting